src/neptune/internal/operation_processors/operation_storage.py

```diff
@@ -55,6 +56,11 @@ def upload_path(self) -> Path:
         return self.data_path / "upload_path"
 
     def cleanup(self) -> None:
+        disable_cleanup = os.getenv(NEPTUNE_DISABLE_LOCALFILES_CLEANUP, "false").lower() in ("true", "1", "t")
+        if not disable_cleanup:
+            self._remove_storage_folder()
+
+    def _remove_storage_folder(self):
         shutil.rmtree(self.data_path, ignore_errors=True)
```

Comment:
This cannot be impacted by flag, only a few lines below that impact parent directory